crates/synthesis/src/linalg/cos_sin_decomp.rs

```diff
@@ -120,12 +75,10 @@ pub fn cos_sin_decomposition(u: DMatrix<Complex64>) -> CosSinDecompReturn {
     // We will first find l0, c, and r0, then s and l1, and finally r1.
 
     // Apply SVD to u00
-    let svd = u00
-        .try_svd(true, true, 1e-12, 0)
-        .expect("Problem with SVD decomposition");
-    let mut l0 = svd.u.unwrap();
-    let mut r0 = svd.v_t.unwrap();
-    let mut c: DVector<f64> = svd.singular_values.column(0).into_owned();
+    let svd = u00.svd().expect("Problem with SVD decomposition");
```

Comment:
do you want to add `verify_svd_decomp` (or a similar code in faer): https://github.com/Qiskit/qiskit/blob/544885d788e51acbc84c7cbd85510232da980561/crates/synthesis/src/linalg/mod.rs#L44
similar to what we did in:
https://github.com/Qiskit/qiskit/blob/429a9c0eaf5f77ee5b22a7601d4acf2df7d2c4c1/crates/synthesis/src/linalg/mod.rs#L122